[FE] 2주차 리뷰

choichoigang opened the pull request:
### 상단 숙소 필터링이 가능한 메뉴 구현 , 숙소 리스트 스타일링

- 상단 숙소를 요구 조건에 맞춰서 필터링이 가능하도록 구현
- 하단 숙소 스타일링을 구현
- Airbnb 라이브러리 활용

choichoigang committed:
[#9] Feat : npm install :  css-loader , moment  , react-dates , react-redux , react-route-dom , redux , style-loader , styled-components , styled-reset

choichoigang committed:
[#9] Docs : css-loader , style-loader rule 추가

choichoigang committed:
[#9] Docs : 산돌 폰트 적용

choichoigang committed:
[#9] Feat : index.js Router Component 적용 , yarn file update

choichoigang committed:
[#9] Docs : Class Component -> function Component로 변경

choichoigang committed:
[#9] Feat : Redux Ducks 구조 파일 추가

choichoigang committed:
[#9] Feat : pretterrc 추가

choichoigang committed:
[#9] Feat : Component 기본 구조 추가

choichoigang committed:
Merge pull request #13 from codesquad-member-2020/feature/fe-environment

Feature/fe environment

choichoigang committed:
[#14] Feat : 상단 Nav Component 구현

choichoigang committed:
[#14] Docs : index.html title 변경 및 App.js nav 영역 추가

choichoigang committed:
Merge pull request #15 from codesquad-member-2020/feature/fe-nav

Feature/fe nav

Elllin committed:
[#16] person action creator, reducer 생성

Elllin committed:
[#16] feat: store생성 후 provider로 넣어 줌

Elllin committed:
[#16] feat: 인원 토글 버튼

클릭 시 인원창 나오고 다시 클릭 시 인원창 사라진다

Elllin committed:
[#16] feat: Person Component 기본 style 및 Click Event 부분 적용

choichoigang committed:
[#16] Docs : Global Style , Custom Style 일부 추가

choichoigang committed:
[#16] Feat : npm redux-devtools-extension 추가 및 적용

choichoigang committed:
[#16] Docs : 폴더 및 파일명 person -> Personnel로 변경

choichoigang committed:
[#16] Feat : person -> personnel로 파일명 및 reducer 함수 변경

choichoigang committed:
[#16] Docs : Global , custom style 관련 구성

choichoigang committed:
[#16] Docs : 스타일 속성 수정 및 prettierrc 속성 일부 변경

choichoigang committed:
[#16] Feat : Personnel Component 수정 및 Personnel Modal 인원 선택 기능 추가

choichoigang committed:
[#16] Docs : useEffect에서 실행한 로직 일반 함수로 분류

choichoigang committed:
[#16] Feat : Personnel의 Type와 Type에 따른 나이 범위를 지정하는 속성 props로 전달 및 style 지정

choichoigang committed:
[#16] Feat : 지우기 button과 저장 button을 지정하는 bottom Area 지정 및 style 속성 추가

choichoigang committed:
[#16] Feat : GlobalStyle 일부 속성 변경 및 Date Component에 대한 CustomStyle 지정

choichoigang committed:
[#16] Feat : Date Component 추가 및 Header Component에 import문 추가

choichoigang committed:
[#16] Docs : SaveButton font-size 및 date box shadow 속성 추가

choichoigang committed:
[#16] Docs : Nav airbnb 로고 추가 및 padding 속성 일부 수정

choichoigang committed:
[#16] Docs : prettierrc 수정

choichoigang committed:
[#16] Feat : react-dates 옵션 추가 및 캘린더에 render될 CalendarButtons Component 추가

choichoigang committed:
[#16] Feat : 경로 수정 및 일부 CSS 속성 CustomStyles로 전역화

choichoigang committed:
[#16] Feat : Date에 대한 Reducer 파일 로직 구현

choichoigang committed:
[#16] Feat : rootReducer에 dateReducer 추가 및 personnel 일부 로직 주석 처리

choichoigang committed:
[#16] Fix : 오류 코드 수정

Elllin committed:
[#16] useFetch 커스텀 hooks로 분리, 초기 렌더링 fetch작업

choichoigang committed:
[#16] Feat : CustomStyle saveButton resetButton 일부 속성 수정

choichoigang committed:
[#16] Docs : 주석 삭제

choichoigang committed:
[#16] Feat : CalendarButtons 전달 방식 일부 수정

choichoigang committed:
[#16] Docs : 디버깅용 코드 일부 주석처리

choichoigang committed:
Merge pull request #21 from codesquad-member-2020/feature/fe-personnel

[FE] Personnel Component , Date Component 구현

choichoigang committed:
[#22] Feat : rheostat lib 설치

choichoigang committed:
[#16] Feat : Price Component 기본 구성 구현 및 CustomStyle 일부 속성 변경

Elllin committed:
[#22] price state 구성, index에서 combine

Elllin committed:
[#22] priceModal 기본 width, height 구성

Elllin committed:
[#22] priceRange rheostat 라이브러리 기본 구성 및 min, max값 지정

Elllin committed:
[#22]feat: 슬라이드 움직일 때 최저요금, 최고요금 반영

Elllin committed:
[#22] refactor: modal 지우기, 저장 버튼 개별 컴포넌트로 분리

Elllin committed:
[#22] feat: resetHandler 기능 구현

리셋버튼 클릭 시 초기화

choichoigang committed:
[#22] Feat : Chart Component 추가 및 일부 속성 수정 , 테스트용 상수 추가

choichoigang committed:
[#22] Feat : ChartComponent  , ChartBarComponent 추가

choichoigang committed:
[#22] style : PriceRange lib style overriding

choichoigang committed:
[#22] Feat : Server에 전달 방식에 맞춰서 초기 데이터 변경

choichoigang committed:
[#22] Feat : 로그용 코드 수정

choichoigang committed:
[#22] Feat : props로 전달한 widthPercent로 ChartBar의 크기를 정할 수 있도록 전달

choichoigang committed:
[#22] Docs : setChartBarHeight -> analyseChartData 로 함수명 변경 및 calculationChartDataLocate 인자 명 변경

choichoigang committed:
[#22] Docs : 주석 코드 제거

choichoigang committed:
[#22] Feat : 기존 Array 전달 형식에서 Number Type의 chartBarCount를 전달하면 전달한 수 만큼의 ChartBar를 만들어 주는 형식으로 변경

choichoigang committed:
[#22] Feat : prop-types을 이용해서 전달받는 PropTypes을 검사해주는 로직 추가

Elllin committed:
Merge pull request #30 from codesquad-member-2020/feature/fe-price

Elllin committed:
[#29] feat: 초기 url에 오늘, 내일 날짜 파라미터로 전달

오늘, 내일 날짜 구하는 함수구현
구한 날짜를 파라미터에 전달

Elllin committed:
[#29] feat: rooms title 스타일 추가, fetch data 구조분해할당

폰트사이즈및 layout

Elllin committed:
[#29] feat : 숙소리스트 컴포넌트 분리, 숙소이미지 렌더링

숙쇠리스트들을 렌더링하는 컴포넌트 분리

choichoigang committed:
[#22] Feat : Chart Component의 Props 로 전달되는 값의 기본을 지정

choichoigang committed:
[#22] Feat : Chart의 defaultProps 지정 및 Chart의 PropTypes의 속성 지정

choichoigang committed:
[#22] Docs : 로깅용 데이터 삭제 및 포인트 지정

Elllin committed:
[#29] 숙소 content 지역, 별점 스타일 적용

별아이콘 폰트어썸 설치 및 import

choichoigang committed:
[#22] Feat : npm 추가

Elllin committed:
[#29] 전체 wrap 사이드 padding 추가 및 margin 추가

Elllin committed:
[#29]feat:이미지 hover시scale 효과 적용

Elllin committed:
[#29] feat: 요금, 예약버튼 스타일

요금 숫자 3자리 콤마찍는 함수구현
할일금액 없을 시 예외처리 미구현

Elllin committed:
Merge pull request #31 from codesquad-member-2020/feature/fe-roomsList

choichoigang committed:
[#22] Feat : Custom Style 변경 및 webpack 설정 일부 변경

choichoigang committed:
[#22] Feat : PriceBox 스타일 수정

choichoigang committed:
[#22] Docs : 일부 속성 변경

choichoigang committed:
[#22] Feat : ChartBar css 일부 속성 변경

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-price

# Conflicts:
#	FE/airbnb/package.json

Elllin committed:
[#29] feat: 가격 할인아닐 시 할인 아닌 가격만 보여주도록 수정

choichoigang committed:
Merge pull request #32 from codesquad-member-2020/feature/fe-price

Feature/fe price

Elllin committed:
[#29] refactor: 숙소리스트 width px -> %로 변경

choichoigang committed:
[#22] Feat : ChartBar의 증가 단위에 대한 변수명 변경 chartBarIncreasePoint -> chartBarIncreaseUnit , ChartData의 값이 사용자가 정한 범위를 초과하는 경우에는 마지막 ChartBar의 Height 크기를 증가시키도록 조건문 추가

choichoigang committed:
Merge pull request #36 from codesquad-member-2020/feature/fe-price

Feature/fe price

choichoigang committed:
[#36] Feat : Price Component에 Modal의 외부 영역을 Click했을 때 Modal을 Close하는 Event 추가

choichoigang committed:
[#33] Feat : Props 전달 구조 변경

기존 상위 Component Price에서 Redux Store에 필요한 값들을 Props로 전달하는 구조에서 , Store에 값이 필요한 Component들이 직접 UseSelector와 UseDispatch를 통해서 값을 사용하도록 변경

Elllin committed:
[#29] refactor: 총 요금 color 블랙->회색으로 변경

창 크기 줄일 시 width 비율및 양옆 간격 유지하도록 %로 변경 및 margin 추가

choichoigang committed:
[#33] Feat : ModalButtons Component Style 수정

ModalButtons Component 사용시 width와 height 값을 직접 설정할 수 있는 기능 추가

choichoigang committed:
[#33] Feat : Personnel Component 하단 ButtonArea를 ModalButtons Component로 변경

choichoigang committed:
[#33] Docs : resetEvnet function 함수명 변경 resetSelectPersonnel -> onClickResetHandler

choichoigang committed:
[#33] Feat : Personnel Component Outside Click Event 추가

Personnel의 Modal이 active된 상태에서 Modal 영역의 밖을 클릭하면 Modal이 deactivate되도록 변경

choichoigang committed:
[#33] Feat : Date의 CalenderButtons Component를 ModalComponent로 변경 및 CalenderButtons Component 삭제,  Personnel Component 불필요한 스타일 삭제

choichoigang committed:
[#33] Feat : Chart Component에 전달된 props에 따라서 Chart의 Width를 정할 수 있는 기능 추가

choichoigang committed:
[#33] Feat : ChartBar 단위 변경 , Chart 선택 범위에 따른 색상 표시 Component 추가

ChartBarIncreasePoint px 단위에서 % 단위로 변경
ChartBar 선택 범위에 따라서 색상을 나타내는 ChartStartPoint , ChartEndPoint 추가 및 일부 style 속성 변경

choichoigang committed:
[#33] Feat : chartGauge 속성을 전달해서 Chart의 선택된 mix max의 영역을 보여주는 기능을 활성화 할 것인지를 정하는 로직 추가

choichoigang committed:
[#33] Feat : Chart Component Gauge 관련 로직 수정

 Component :ChartGauge의 start 와 end value를 props로 전달받는 방식으로 변경 (chartGaugeStart , chartGaugeEnd) ,
 calculationGaugeWidth : 전달받은 Gauge값을 계산해주는 함수 추가

choichoigang committed:
[#33] Feat : Chart Component 사용자가 직접 height value를 props로 전달해서 변경할 수 있도록 로직 변경

choichoigang committed:
[#33] Docs : 불필요한 import문 제거

Elllin committed:
[#29] refactor: 조건문 뒤에 render함수 분리

Elllin committed:
Merge pull request #37 from codesquad-member-2020/feature/fe-roomsList

Elllin committed:
[#29] fix: 예약 버튼 onClick 빈 이벤트 삭제

Elllin committed:
Merge pull request #38 from codesquad-member-2020/feature/fe-roomsList

[fix] onClick 오류 수정

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-modalEvent

choichoigang committed:
Merge pull request #41 from codesquad-member-2020/feature/fe-modalEvent

Feature/fe modal event

Elllin committed:
[#39] feat: 예약 버튼 클릭시 모달창 open

isClicked state를 reservation에서 관리

choichoigang committed:
[#42] Feat : createChartBar function for문 로직을 array.from으로 배열화 시킨 후 map method 활용한 로직으로 변경

Elllin committed:
[#39] feat: 모달창 위치 layout

Elllin committed:
[#39] feat: 예약버튼 클릭 시 그림자 처리

choichoigang committed:
[#33] Feat : today 기준 년월일 , tomorrow 기준 년월일을 구하는 util 함수 추가

Elllin committed:
[#39] feat: close버튼 클릭 시 모달창 close

choichoigang committed:
[#33] Feat : Date Component 스타일 수정

choichoigang committed:
[#33] Feat : DatePickerInput 영역을 Click 했을 때 checkInDate가 StartDate로 고정되도록 수정

choichoigang committed:
[#33] Feat : npm install

choichoigang committed:
[#33] Feat : Dotenv를 사용해서 환경 변수로 API 관리

choichoigang committed:
[#33] Feat : Date Compoent 불필요한 import문 및 props option 삭제

crongro commented on FE/airbnb/src/App.js:
```diff
@@ -0,0 +1,26 @@
+import { GlobalStyle } from "./style/GlobalStyle.jsx";
+import styled from "styled-components";
+
+const App = () => {
```
엔트리 포인트에서 이렇게 구조를 잘 표현해주니 좋습니다.

crongro commented on FE/airbnb/src/App.js:
```diff
@@ -0,0 +1,26 @@
+
+import Rooms from "./components/rooms/Rooms.jsx";
+
+import { GlobalStyle } from "./style/GlobalStyle.jsx";
```
중복 css를 제거하는 효과적인 방법이죠.
중복은 최대한 제거해서 잘 관리하도록 하시면 좋죠. 이게 쉽지는 않아보이더라고요.

crongro commented on FE/airbnb/src/components/header/date/Date.jsx:
```diff
@@ -0,0 +1,70 @@
+import "react-dates/initialize";
+import "react-dates/lib/css/_datepicker.css";
+import "moment";
+import "moment/locale/ko";
```
라이브러리를 사용할때, 장단점을 배우고, 라이브러리의 구조도 배우는게 좋아요.

그래야 오픈소스 수준은 아니더라도, 나만의, 또는 회사안에서 사용가능한 라이브러리를 만들때 아이디어를 배울 수 있죠.

지금도 비슷하게라도 작은 라이브러리를 만들어서, 흉내내는 경험 필요하고요.

crongro commented on FE/airbnb/src/components/header/date/Date.jsx:
```diff
@@ -0,0 +1,70 @@
+          endDateId="tata-end-date"
+          onDatesChange={handleDatesChange}
+          focusedInput={focusedInput}
+          onFocusChange={(focusedInput) => {
```
bind의 커링장점을 활용해서 요렇게도 표현될까요?

onFocusChange={setFocusedInput.bind(undefined, focusInput)}

crongro commented on FE/airbnb/src/components/header/personnel/Personnel.jsx:
```diff
@@ -0,0 +1,109 @@
+
+import {
+  increaseAdultCount,
+  decreaseAdultCount,
```
action 의 이름들이 좋네요.

crongro commented on FE/airbnb/src/components/header/personnel/Personnel.jsx:
```diff
@@ -0,0 +1,109 @@
+  };
+
+  useEffect(() => {
+    window.addEventListener("click", onClickOutsideHandler);
```
이거 매번 컴포넌트 렌더링될때마다 실행되는거 같은데 맞나요?

crongro commented on FE/airbnb/src/components/header/personnel/Personnel.jsx:
```diff
@@ -0,0 +1,109 @@
+            onIncrease={increaseChildCount}
+            onDecrease={decreaseChildCount}
+            personnelType="어린이"
+            personnelScope="2 ~ 12세"
```
jsx안에서 이런 문자열도 '성인', '어린이', '2~12세등'을 없앨 수도 있을까요?
jsx가 저런 문자열 변경에 영향을 안받으면 더 좋을 듯.

crongro commented on FE/airbnb/src/components/header/personnel/PersonnelTabColumn.jsx:
```diff
@@ -0,0 +1,83 @@
+  );
+};
+
+const MIN_COUNT = 0;
```
이값을 PersonnelTabColumn 함수밖에 선언한것은 이해하고요.
PersonnelTabColumn 함수 위쪽에 선언하는게 어때요? 
이름은 조금더 구체적으로 짓고요.
```
const PERSON_COUNT = {
   MAX : 8, MIN : 0
}
```

crongro commented on FE/airbnb/src/components/header/price/Price.jsx:
```diff
@@ -0,0 +1,40 @@
+import PriceModal from "./PriceModal.jsx";
+
+import styled from "styled-components";
+import { Button } from "../../../style/CustomStyle.jsx";
```
나중에 path가 복잡해지면 webpack 의 alias 를 활용해서 줄여보시죠.

crongro commented on FE/airbnb/src/components/header/price/PriceBoxes.jsx:
```diff
@@ -0,0 +1,65 @@
+const PriceBox = styled.div`
+  width: 14vw;
+  height: 50px;
+  border: 1px solid var(--gray-1);
```
변수활용 좋아요.

crongro commented on FE/airbnb/src/components/header/price/PriceModal.jsx:
```diff
@@ -0,0 +1,73 @@
+
+const PRICE_UNIT = 50000;
+const CHART_COUNT = 20;
+const CHART_DATAS = [
```
지금이야 mock data로 이렇게 표현했지만, 
서버에서 받은 값으로 표현하면 불필요한 코드겠군요.

crongro replied:
mock으로 표현할때도 함수로 만들어도 좋을 듯.
예를들어서 getData라는 함수를 별도 파일로 분리해서 만들고 아래처럼 mock환경인지 확인해서 처리한다던가.
실제 서버와 통신할때는 getData를 false로 줄 수 있도록 어떻게 잘 설정한다던가 해서 PriceModal.jsx에서는 저런 mock코드를 지웠다 추가했다 하는 일을 없애는 거죠. 

```
function getData({mock=true}) {
 if(mock) return [5000,5000,,,,,,400000];
}
```

crongro commented on FE/airbnb/src/components/header/price/PriceRange.jsx:
```diff
@@ -0,0 +1,65 @@
+  const dispatch = useDispatch();
+
+  const onValuesUpdatedHandler = ({ values }) => {
+    dispatch(setPrices(values));
```
values 보다는 의미있는 값으로 이름을 지어보세요.
코드에서 네이밍은 모두 의미를 가지고 있어야 합니다.

crongro commented on FE/airbnb/src/components/header/price/chart/Chart.jsx:
```diff
@@ -0,0 +1,178 @@
+        />
+      );
+
+      chartBars.push(chartBar);
```
이 코드는 map을 활용해서 리팩토링 가능해보이네요.

crongro commented on FE/airbnb/src/components/header/price/chart/Chart.jsx:
```diff
@@ -0,0 +1,178 @@
+  );
+};
+
+Chart.propTypes = {
```
이런 시도 좋고, ts와 약간 중복되는 장점을 가진 것이죠.
ts를 대체하는건 아니지만  ts를 사용하지 않는다면,  컴포넌트에서 이거 사용해도 좋아요.

crongro commented on FE/airbnb/src/components/rooms/Rooms.jsx:
```diff
@@ -0,0 +1,74 @@
+  let month = today.getMonth() + 1;
+  const year = today.getFullYear();
+
+  if (day < 10) day = `0${day}`;
```
10 이런 매직넘버 없애세요~

crongro commented on FE/airbnb/src/modules/date.js:
```diff
@@ -0,0 +1,43 @@
+// action type
+const START_DATE = "date/START_DATE";
```
action / reducer 코드가 깔끔해서 좋네요. 👍

crongro reviewed (approved):
잘 구현했어요. 👍 

리뷰내용 참고해서 이어서 개발하시죠!

Elllin committed:
[#39] feat: 초기 데이터 fetch

예약하기 버튼 클릭 시 fetch작업

Elllin committed:
[#39] feat: fetch 함수 분리, fetch후 data store에 저장

choichoigang committed:
[#33] Feat : setCheckDateQueryParameter funciton 함수 추가

체크인 체크아웃의 선택된 상태에 따라서 store에 값을 Update해주고 Query Parameter 형식에 맞게 string type을 return 해주는 함수 구현

choichoigang committed:
[#33] Feat : setPersonnelQueryString 함수 추가 , setRequestURL 함수 로직 구현

setPersonnelQueryString : 만약 인원이 선택되지 않았다면 default 값 adult=1 을 Query String으로 return 한다.
setRequsetURL : 필터링에 대한 정보를 QueryString으로 바꾸고 DB HOST 주소에 더해서 요청 URL을 생성한다.

choichoigang committed:
[#33] Docs : 불필요한 console문 제거

choichoigang committed:
Merge pull request #48 from codesquad-member-2020/feature/fe-filtering

Feature/fe filtering

choichoigang committed:
[#47] Feat : npm install query-string

choichoigang committed:
[#47] Feat : Component들을 Route 처리

choichoigang committed:
[#47] Docs : useFetch Custom hook에 try catch문 추가

choichoigang committed:
[#47] Feat : ModalButtons Component 내부 함수 수정 및 일부 로직 변경

setCheckDateQueryString : (startDate && !endDate) 조건 내부 로직 수정
createQueryString : setRequsetURL에서 createQueryString으로 변경
saveHandler : fetch 로직에서 history.push를 사용해서 리다이렉트 시키는 로직으로 변경

choichoigang committed:
[#47] Feat : height 의 퍼센트가 100을 넘을시에 더이상 height값을 추가하지 않는 로직 추가

choichoigang committed:
[#47] Docs : reducer destructuring 변경

choichoigang committed:
[#47] Feat : Chart Component에 전달하는 Props를 확인한 후 error를 처리해주는 checkEssentialPropValue 함수 생성

요소를 구성하는데 기본적인 props를 전달받지 못하면 Error를 throw 한다.
차트의 증가 단위가 조건을 통과하지 못하면 불완전한 그래프가 그려질 수 있음을 알리는 error를 출력한다.

choichoigang committed:
[#47] Feat : 불필요한 console.log 제거

Elllin committed:
[#39] feat: 버튼 클릭 시 id에 맞는 데이터 받도록 url에 id값 삽입

Elllin committed:
[#36] refactor: 필요없는 콘솔로그  제거

Elllin committed:
[#39] feat: 1박당 가격 렌더링,평점(별)  아이콘 추가

Elllin committed:
[#39] feat: 숙소당 별점 숙소리스트에서 데이터를 store에 저장

Elllin committed:
[#39] feat: 성능최적화를 위해 memo추가 숙소 list mock data 추가

숙소 layout width넘어가면 아래로 흐르도록 flex-wrap추가

Elllin committed:
[#39] feat: socre 스타일 추가

choichoigang committed:
[#47] fix : CheckOutDate 오류 수정

choichoigang committed:
Merge pull request #50 from codesquad-member-2020/feature/fe-pagination

Feature/fe-pagination

Elllin committed:
[#39] feat: 인원수 select box 추가

option은 map으로 렌더링함수 구현

Elllin committed:
[#39] feat: 각종 요금 렌더링 및 스타일링

Elllin committed:
[#39] feat: 예약하기 버튼 custom으로분리, 예약하기 하단 문구 생성

Elllin committed:
[#39] refactor: 매직넘버 제거

Elllin committed:
Merge pull request #51 from codesquad-member-2020/feature/fe-reservation

choichoigang committed:
[#47] Fix : setCheckDateQueryString 오류 로직 수정

Elllin committed:
[#39] fix: key값 없는 오류 수정, date잘못 사용 오류 수정

Elllin committed:
[#36] refactor: 통화 구하는 함수 유틸로 분리, 예약하기 모달창 전체
padding 조절

Elllin committed:
[#39] refactor: mock대신api를 이용해 fetch작업

Elllin committed:
[#39] refactor: 모달창 띄울 때 쉐도우 조정

Elllin committed:
[#39] feat: 인원 선택시 선택된 인원만큼 select되도록 설정

Elllin committed:
[#39] refactor: select value속성을 줘서 인원 선택화면 봐뀌도록 수정

Elllin committed:
[#39] refactor: 모달창 열리고 닫힐때 useState로 사용

choichoigang committed:
[#47] Docs : .env ROOMS_DB_HOST 주소 변경

choichoigang committed:
[#47] Feat : 초기 requsetURL 생성 함수 추가

choichoigang committed:
[#47] Feat : fetch로직 및 getDate unit과 custom hook으로 분리 및 일부 로직 삭제

choichoigang committed:
[#47] Feat : useRoomsFetch Custom hook 생성

초기에 필터링을 설정하지 않았을 시 , 새로고침이 될 때 , 필터링 옵션이 변경될 때를 분기 처리 해서 fetch url을 변경시키는 Custom hook 추가

choichoigang committed:
[#47] Feat : personnel Type에 따라서 데이터를 set 해주는 reducer 추가

choichoigang committed:
[#47] Feat : 화면이 새로고침 될 때 query string 에 담겨있는 값의 여부에 따라서 store에 업데이트 해주는 식 추가

choichoigang committed:
[#47] Feat : Personnel Reduer 수정

increase , decrease action을 parameter에 personnelType을 전달해서 하나의 함수로 재사용 가능하도록 구조 변경
유아의 경우에는 totalCount에 추가 시키지 않는 로직으로 변경
변경된 personnel reducer에 따라서 Personnel Component 들 내부 로직 수정

choichoigang committed:
[#47] Docs : price Number Type으로 전달

Elllin committed:
[#39] feat: fetch 파라미터 인원 반영

Elllin committed:
Merge pull request #53 from codesquad-member-2020/feature/fe-reservation

choichoigang committed:
[#47] Feat : server에서 오는 가격 정보 기본 값 추가

choichoigang committed:
[#47] Feat : 현재 동작하지 않는 코드 주석처리

choichoigang committed:
[#47] Feat : MockData 제거 후 현재 Server 기준으로 변하는 Date로 변경

choichoigang committed:
[#47] Docs : Chart Compoent 매직 넘버 제거

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

# Conflicts:
#	FE/airbnb/.env.development
#	FE/airbnb/src/components/rooms/Rooms.jsx

choichoigang committed:
Merge pull request #54 from codesquad-member-2020/feature/fe-pagination

Feature/fe pagination

choichoigang committed:
[#47] Feat : 초기 url 받아오는 함수 제거 및 console문 제거

choichoigang committed:
[#47] Feat : Reset Action Function 추가

Elllin committed:
[#39] refactor: 기간당 요금 렌더링 추가, 모달창 padding 조절

choichoigang committed:
[#47] Feat : Logo를 클릭시 root 주소로 이동시킨 후 새로고침을 하는 로칙 추가

choichoigang committed:
[#47] Feat : import문 Convention 수정 및 personnel이 선택되지 않았을 때 adult Store를 update 시켜주는 로직 추가

Elllin committed:
[#38] refactor: 기간 반영 useState -> 변수 사용하여 렌더링

Elllin committed:
[#39] fix: 초기 선택된 date없을때 오늘과 내일 날짜로 조회

Elllin committed:
[#39] feat: 파라미터 값 가져와서 예약하기 클릭 시 fetch

Elllin committed:
[#39] refactor: 불필요한 console제거

Elllin committed:
[#39] feat: 예약하기 버튼 클릭시 확인 후 예약진행

Elllin committed:
Merge pull request #57 from codesquad-member-2020/feature/fe-reservation

Elllin committed:
[#58] feat: 페이지네이션 reducer 생성 및 초기값 생성

Elllin committed:
[#58] feat: action 생성 함수 생성

Elllin committed:
[#58] feat: 페이지네이션 배열 생성 및 화면에 렌더링

Elllin committed:
[#58] feat: 페이지 번호 클릭시 현제 페이지를 state에 업데이트

choichoigang committed:
[#47] Docs : 불필요한 console 문 제거

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

# Conflicts:
#	FE/airbnb/src/components/rooms/RoomsList.jsx

choichoigang committed:
[#47] Feat : Custom Style ToggleWrap 속성 변경

choichoigang committed:
[#47] Docs : 인원 현황 totalCount로 변경

choichoigang committed:
[#47] Feat :useRoomsFetch setQueryString 조건식을 함수화 한 후 useEffect에서 선헌

choichoigang committed:
[#47] Feat : react-dates 라이브러리 스타일 일부 수정

choichoigang committed:
[#47] Feat : 코드 컨벤션 수정

Elllin committed:
[#58] refactor: params변경 하는 로직 구현

choichoigang committed:
[#47] Feat : pagination이 발생하면 roomsList Store에 accommodations의 값을 최신으로 업데이트 시켜주는 로직 구현

Elllin committed:
Merge branch 'feature/fe-pagination' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

Elllin committed:
Merge pull request #60 from codesquad-member-2020/feature/fe-pagination

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

choichoigang committed:
[#47] Feat : 예약창을 클릭시 현재 screen의 height와 offsetY값을 구해서 중앙 정렬 시키는 로직으로 변경

choichoigang committed:
[#47] Docs : 중복된 useSelector 하나의 식에서 사용 가능하게 변경

Elllin committed:
[#58] feat: 페이지네이션 끝번호에 갔을때 다음 페이지번호 보여주기

Elllin committed:
[#58] refactor: 다음 페이지네이션 보여주기 함수 중복코드 제거

Elllin committed:
[#58] feat: 가장 처음과 가장 마지막 인덱스로 가는 버튼 구현

Elllin committed:
[#58] fix: 페이지네이션 key값 없는 오류 수정, 숙소 타이틀 높이값 수정

Elllin committed:
[#58] refactor: 예약하기 모달창 스크롤 내려도 가운데 위치하도록 수정

choichoigang committed:
[#47] Feat : Util 함수 추가

createQueryString : 초기 요청에 필요한 QueryString 을 생성해주는 함수
setCurrentPage : Query에 있는 OffsetItems의 값을 받아서 현재 CurrentPage의 Number를 return 하는 함수 구현

choichoigang committed:
[#47] Feat : OffsetItems가 존재할 경우 CurrentPage에 dispatch 해주는 && 연산자 추가

choichoigang committed:
[#47] Feat : onClickPage 이벤트가 발생할 시 조건에 따라서 rooms Component로 Push 하는 로직 추가

choichoigang committed:
[#47] Feat : 사용하지 않는 props 제거

Elllin committed:
[#58] refactor: 페이지 인덱스 변수명 변경

choichoigang committed:
[#47] Feat : Page 전환이 일어나면 스크롤을 최상단으로 이동시켜주는 유틸 함수 추가 및 Pagination onClickPage에 적용

choichoigang committed:
Merge pull request #61 from codesquad-member-2020/feature/fe-pagination

Feature/fe pagination

Elllin committed:
[#58] refactor: 페이지 버튼을 다른 컴포넌트로 분리

Elllin committed:
[#58] refactor: 이전, 다음 버튼 폰트어썸 아이콘 추가

페이지 번호 렌더로직 함수로 분리

Elllin committed:
[#58] feat: 페이지 번호 layout 및 padding, margin 추가

Elllin committed:
[#58] fix: 마지막 인덱스로 이동후 이전 버튼 클릭 시 오류 수정

choichoigang committed:
[#47] Feat : fetch 시에 Page button이 클릭되면 roomsList만 dispatch 하기위해서 paginationActive 값과 action 함수 추가

choichoigang committed:
[#47] Feat : 페이지 버튼이 눌렸을 때 pagination active를 ture로 바꿔주는 로직 추가

choichoigang committed:
[#47] Feat : accommodations만 update해주는 Action 추가

choichoigang committed:
[#47] Feat : Action Function Parameter 전달 형식에서 useFetch 폴더에서 직접 부르는 방식으로 변경 및 fetchRooms에 조건식 추가

choichoigang committed:
[#47] Docs : 화면 기준 중앙 계산식 수정 window.screen.height -> window.innerHeight

choichoigang committed:
Merge pull request #62 from codesquad-member-2020/feature/fe-pagination

Feature/fe pagination

Elllin committed:
[#58] feat: 번호 클릭 시 포커스 효과, hover시 포커스 효과

Elllin committed:
[#38] feat: 버튼 쉐도우, 컬러 스타일

Elllin committed:
[#39] 페이지 번호 클릭 시 최상단으로 이동

Elllin committed:
[#58] feat: 이전, 다음 버튼 클릭 으로 이동시 페이지 최상단으로 이동

Elllin committed:
[#58] refactor: logo크기 키움

Elllin committed:
[#58] refactor: 주석제거

Elllin committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

Elllin committed:
[#58] feat: 필터링 저장 버튼 클릭시 1페이지로 이동

Elllin committed:
[#58] feat: 새로고침했을 때 현재 페이지 numbers 배열 가져와 렌더링

Elllin committed:
[#58] refactor: 예약하기 모달창 금액레이아웃 컴포넌트 분리

Elllin committed:
[#58] fix: 조건 필터링 할때 첫번째 페이지로 이동하지 않는 오류 수정

Elllin committed:
[#58] refactor: console제거 및 삼항연상자 정리

Elllin committed:
[#58] refactor: classname직접 지정하지 않고props로 받아서 처리

Elllin committed:
[#58] refactor: console 제거

Elllin committed:
Merge pull request #63 from codesquad-member-2020/feature/fe-pagination

Elllin committed:
[#64] refactor: room 컴포넌트에서 roomlist 렌더링 함수 분리

Elllin committed:
[#64] fix: 1페이제이서 이전가기버튼클릭시 오류 수정

choichoigang committed:
[#47] Feat : 저장 버튼 클릭시 현재 Active된 modal이 닫히는 이벤트 추가

Elllin committed:
[#64] fix: 10단위 페이지에서 끝으로 가기 버튼 클릭 시 오류 수정

choichoigang committed:
[#47] Feat : Price Modal 평균 숙박 금액 Content 추가 및 npm install react-slick

Elllin committed:
[#64] feat: 예약하기 버튼 클릭시 최종 예약 fetch

Elllin committed:
[#64] feat: 예약 후 페이지 refresh

choichoigang committed:
[#47] Feat : slick-carousel 관련 css link 추가

choichoigang committed:
[#47] Feat : npm install react-slick

choichoigang committed:
[#47] Feat : yarn add

Elllin committed:
[#64]

Elllin committed:
[#64] refactor : console 제거

choichoigang committed:
[#47] Feat : Image Slider 기능 추가

Elllin committed:
Merge pull request #67 from codesquad-member-2020/feature/fe-refactor

choichoigang committed:
[#47] Feat : getCookie util 추가

choichoigang committed:
Merge branch 'fe-dev' of https://github.com/codesquad-member-2020/airbnb-10 into feature/fe-pagination

choichoigang committed:
[#47] Feat : Reservation Modal Shodow Style Error 수정

choichoigang committed:
[#47] Feat : RoomsList 스타일 일부 수정

choichoigang committed:
Merge pull request #68 from codesquad-member-2020/feature/fe-pagination

Feature/fe pagination
